modules/service-account-auto-secret-removed.adoc

```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// * authentication/using-service-accounts-in-applications.adoc
+// * pods/nodes-pods-secrets.adoc
+
+:_content-type: CONCEPT
+[id="auto-generated-sa-token-secrets_{context}"]
+= About automatically generated service account token secrets
+
+When a service account is created, a service account token secret is automatically generated for it. This service account token secret, along with an automatically generated dockercfg secret, is used to authenticate to the internal {product-title} registry.
+
+[NOTE]
+====
+Prior to {product-title} 4.11, a second service account token secret was generated when a service account was created. This service account token secret was used to access the Kubernetes API.
+
+Starting with {product-title} 4.11, this second service account token secret is no longer created. This is because the `LegacyServiceAccountTokenNoAutoGeneration` upstream Kubernetes feature gate was enabled, which stops the automatic generation of secret-based service account tokens for accessing the Kubernetes API.
+
+After upgrading to {product-version}, any existing service account token secrets are not deleted and continue to function.
+====
+
+If your workload needs a service account token, use the `TokenRequest` API to request a bound service account token, because these tokens are invalidated when their associated pod is deleted. You can also manually create a service account token secret to create a token that never expires.
```

Comment:
> because these tokens are invalidated when their associated pod is deleted.

That's not the reason to use them, not to mention that by default these will not be associated to any pod.

The reasons to use these are:
- they expire with time
- they contain audiences
- they can be bound to pods/secrets and will be invalidated if the bound object is removed.
All the attributes above make them safer to use.

Reply:
> You can also manually create a service account token secret to create a token that never expires.

Should we link the upstream docs on how to do this?

Reply:
@stlaz I found some wording we had elsewhere on why we recommend using the TokenRequest API, so I went with that. Let me know if you want me to still incorporate any of the other reasons you listed here too.

For linking how to do that, we do have this section: https://docs.openshift.com/container-platform/4.13/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating-sa_nodes-pods-secrets. We provide a link below this section (in the additional resources section). I've also added a note here to see that section (link provided below instead of inline unfortunately due to technical limitations).

This section I mention is where I pulled some of this new wording from. Including "if the security exposure of a non-expiring token in a readable API object is acceptable to you". 

Let me what you think after these updates.

Diff of today's changes: https://github.com/openshift/openshift-docs/compare/71b17f9c41b9893cc7a89f0f1bd37b212d8abfa1..51e09e2942da110ba913b753e8f0cd8967b8e464